viewing_party/party.py

```diff
@@ -2,22 +2,178 @@
 
 def create_movie(title, genre, rating):
     pass
+    #make a dictionary with these keys: "title","genre","rating"
+    movie_dict = {}
+    movie_dict["title"]= title
+    movie_dict["genre"] = genre
+    movie_dict["rating"] = rating
+    
+    information = [title,genre,rating]
+    if None in information:
+        return None
+    else:
+        return movie_dict
+
+def add_to_watched(user_data, movie):
+    user_data["watched"] = [movie]
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"] = [movie]
+    return user_data 
+
+def watch_movie(user_data,title):
+    watched_movie = title
+    for movie in user_data["watchlist"]:
+        for value in movie.values():
+            if value == watched_movie:
+                user_data["watchlist"].remove(movie)
+                user_data["watched"].append(movie)
+
+    return(user_data)
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    ratings = []
+    ratings_sum = 0
+
+    if(len(user_data["watched"])) is 0:
+        ratings_average = 0.0
+    else:
+        for i in range(len(user_data["watched"])):
+            ratings.append((user_data["watched"][i]["rating"]))
+            i+=1
+    
+        for rating in ratings:
+            ratings_sum += rating
+        
+        ratings_average = ratings_sum / len(ratings)
+
+    return ratings_average  
+
+def get_most_watched_genre(user_data):
+    genre_list = []
+    genre_dictionary = {}
+
+    if (len(user_data["watched"])) is 0:
+        popular_genre = None
+    else:
+        for i in range(len(user_data["watched"])):
+            genre_list.append((user_data["watched"][i]["genre"]))
+            i += 1
+    
+    for genre in genre_list:
+        count = genre_list.count(genre)
+        genre_dictionary[genre] = count

```

Comment:
nice! this works! but what is happening under the hood in regards to `count`? The built-in function has to iterate through the list to count each genre every single time. And if there are duplicate genres in `genre_list`, then we are repeating and overwriting the same count over and over.

Instead let's add each genre to the `genre_dictionary` inside the for loop on line 63. If the genre has already been added to the dictionary once, then we want to update the count to +1. We could even get rid of `genre_list` altogether!